crates/vm/levm/src/opcode_handlers/system.rs

```diff
@@ -43,11 +43,20 @@ impl VM {
             return Err(VMError::OpcodeNotAllowedInStaticContext);
         }
 
-        let memory_byte_size = (args_offset + args_size).max(ret_offset + ret_size);
+        let memory_byte_size = (args_offset
+            .checked_add(args_size)
+            .ok_or(VMError::VeryLargeNumber)?)
+        .max(
+            ret_offset
+                .checked_add(ret_size)
+                .ok_or(VMError::VeryLargeNumber)?,
+        );
         let memory_expansion_cost = current_call_frame.memory.expansion_cost(memory_byte_size)?;
 
         let positive_value_cost = if !value.is_zero() {
-            call_opcode::NON_ZERO_VALUE_COST + call_opcode::BASIC_FALLBACK_FUNCTION_STIPEND
+            call_opcode::NON_ZERO_VALUE_COST
+                .checked_add(call_opcode::BASIC_FALLBACK_FUNCTION_STIPEND)
+                .ok_or(VMError::OverflowInArithmeticOp)?
```

Comment:
What are the criteria for returning a `VMError::OverflowInArithmeticOp` and `VMError::InternalError`?

Reply:
The Internal's are those that should not happen in any case of execution, and the OverflowInArithmeticOp's are those that should not happen normally.

Reply:
Now I understand, should be an Internal because should not happen in any case